### PR TITLE
docs: add production security checklist

### PR DESCRIPTION
**Pull request by stritti**
## Summary

Adds a comprehensive security hardening checklist for the Pool Controller.

### New Document

- **docs/security-checklist.md** — Security hardening covering:
  - Default password change (web UI + OTA)
  - MQTT authentication and broker isolation
  - LAN-only web interface (no public exposure)
  - Official firmware only policy
  - Backup before OTA updates
  - Flash encryption for advanced setups
  - Network security overview with firewall rules
  - Credential management and storage notes
  - IoT VLAN recommendations

### CODE REVIEW

File: docs/security-checklist.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  - Minimum 8 characters
+  - Mix of uppercase, lowercase, digits, and special characters
+  - Not a reused password from other services
+
+### 2. Change OTA Update Password
+
+The OTA update password is separate from the web UI password.
+
+- **Default**: Same as web UI password
+- **Action**: Web UI → Security & Update → OTA Password
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md]: Fixed in fc361a8. Removed the entire 'Change OTA Update Password' section — the admin password already protects all OTA endpoints via handleAuthentication(). Added a note under section 1 clarifying this.

File: docs/security-checklist.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+- **Recommended**: Create a dedicated MQTT user for the controller with
+  minimal permissions
+
+### 4. Keep MQTT Broker Internal
+
+- **Action**: Bind MQTT broker to local network interface only
+- **Why**: A publicly accessible MQTT broker can be discovered and abused
+- **Mosquitto configuration**:
+  ```
+  listener 1883 192.168.1.0/24
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md]: Fixed in fc361a8. Changed `192.168.1.0/24` to `192.168.1.10` (a real broker IP). Removed the subnet notation since Mosquitto's listener directive only accepts a bind address, not a CIDR range.

File: docs/security-checklist.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+```
+
+### Recommended Firewall Rules
+
+| Source | Destination | Port | Protocol | Action | Purpose |
+|--------|------------|------|----------|--------|---------|
+| Controller | MQTT Broker | 1883 | TCP | Allow | MQTT communication |
+| Controller | NTP Server | 123 | UDP | Allow | Time sync |
+| Controller | Internet | 80 | TCP | Block | No web access needed |
+| Controller | Internet | 443 | TCP | Block | No web access needed |
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md]: Fixed in fc361a8. Changed port 443 from Block to Allow with purpose 'OTA update (Google GitHub)'. Added a note under the table explaining that port 443 can be blocked if OTA is not used.

File: docs/security-checklist.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+For advanced setups:
+
+- Place the ESP32 on an **IoT VLAN** with restricted internet access
+- Allow only: MQTT broker, NTP server, Home Assistant
+- Block all inbound connections from the internet
+
+### 10. Disable Unused Features
+
+- **Action**: If you don't use OTA, disable it in configuration
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md]: Fixed in fc361a8. Replaced the vague 'disable OTA in configuration' with two concrete methods: (1) block outbound HTTPS from the controller at the network firewall, or (2) remove the OTA route registrations in WebPortal.cpp (lines 175-228) and rebuild.

File: docs/security-checklist.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+---
+
+## Related Documents
+
+- [Build from Zero](/docs/build-from-zero/) — Complete build guide
+- [Electrical Safety](/docs/electrical-safety/) — Electrical safety
+- [Safety Model](/docs/safety-model/) — System safety architecture
+- [Production Checklist](/docs/production-checklist/) — Pre-deployment checks
+- [Troubleshooting](/docs/troubleshooting/) — Common issues and fixes
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md]: Fixed in fc361a8. Replaced the 5 dead links with links to existing docs pages: Hardware Guide, Software Guide, MQTT Configuration, and Troubleshooting Matrix (the latter added in PR #101). Also fixed the relay-test procedure reference from the nonexistent Build from Zero to Hardware Guide.

File: docs/security-checklist.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+- Place the ESP32 on an **IoT VLAN** with restricted internet access
+- Allow only: MQTT broker, NTP server, Home Assistant
+- Block all inbound connections from the internet
+
+### 9. Disable Unused Features
+
+- **Action**: If you don't use OTA updates:
+  - Block outbound HTTPS (port 443) from the controller at your network
+    firewall, **or**
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md]: Fixed in fdf36e1. Scoped the HTTPS block advice explicitly: 'To disable GitHub-based OTA only' vs 'To fully disable OTA (including manual upload)'. The latter now requires removing routes and rebuilding, or blocking both inbound and outbound OTA at the firewall.

File: docs/security-checklist.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+### Recommended Firewall Rules
+
+| Source     | Destination | Port | Protocol | Action | Purpose              |
+| ---------- | ----------- | ---- | -------- | ------ | -------------------- |
+| Controller | MQTT Broker | 1883 | TCP      | Allow  | MQTT communication   |
+| Controller | NTP Server  | 123  | UDP      | Allow  | Time sync            |
+| Controller | Internet    | 80   | TCP      | Block  | No web access needed |
+| Controller | Internet    | 443  | TCP      | Allow  | OTA update (GitHub)  |
```
[the remark this answers is not in the document]
Comment (stritti) [reply docs/security-checklist.md:158]: Fixed in fdf36e1. Added 'Controller | DNS Server | 53 | UDP/TCP | Allow | Name resolution' as the first rule in the firewall table. Also applied to German translation.
